src/main/java/burp/BurpExtender.java

```diff
@@ -0,0 +1,356 @@
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+
+import javax.swing.Icon;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JSeparator;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+import javax.swing.SwingUtilities;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.codedx.burp.ContextMenuFactory;
+import com.codedx.burp.ExportActionListener;
+import com.codedx.burp.JTextFieldSettingFocusListener;
+import com.codedx.burp.security.SSLConnectionSocketFactoryFactory;
+
+import jiconfont.icons.FontAwesome;
+import jiconfont.swing.IconFontSwing;
+
+public class BurpExtender implements IBurpExtender, ITab {
+	public IBurpExtenderCallbacks callbacks;
+	private JScrollPane pane;
+
+	private JTextField serverUrl;
+	private JTextField apiKey;
+	private JTextField targetUrl;
+	private JComboBox<NameValuePair> projectBox;
+	private NameValuePair[] projectArr = new BasicNameValuePair[0];
+	
+	public static final String SERVER_KEY = "cdxServer";
+	public static final String API_KEY = "cdxApiKey";
+	public static final String TARGET_KEY = "cdxTarget";
+	
+	@Override
+	public void registerExtenderCallbacks(final IBurpExtenderCallbacks callbacks) {
+		// keep a reference to our callbacks object
+		this.callbacks = callbacks;
+
+		callbacks.registerContextMenuFactory(new ContextMenuFactory(this, callbacks));
+		
+		// set our extension name
+		callbacks.setExtensionName("Code Dx");
+
+		IconFontSwing.register(FontAwesome.getIconFont());
+		
+		// create our UI
+		SwingUtilities.invokeLater(new Runnable() {
+			@Override
+			public void run() {
+				pane = new JScrollPane(createMainPanel());
+
+				callbacks.customizeUiComponent(pane);
+
+				// add the custom tab to Burp's UI
+				callbacks.addSuiteTab(BurpExtender.this);
+			}
+		});
+	}
+
+	private JPanel createMainPanel() {
+		JPanel main = new JPanel();
+		main.setLayout(new GridBagLayout());
+		
+		// Create Settings Panel
+		JPanel settings = new JPanel(new GridBagLayout());
+
+		createTitle("Settings", settings);
+		KeyListener projectEnter = new KeyListener(){
+			@Override
+			public void keyPressed(KeyEvent k) {
+				if(k.getKeyCode() == KeyEvent.VK_ENTER)
+					updateProjects();
+			}
+			@Override
+			public void keyReleased(KeyEvent k) {
+			}
+			@Override
+			public void keyTyped(KeyEvent k) {
+			}
+		};
```

Comment:
FYI, many listener-ish interfaces in java will have a corresponding "adapter" or "default" version of themselves which will provide no-op implementations of each method. Looks like you could use [KeyAdapter](https://docs.oracle.com/javase/8/docs/api/java/awt/event/KeyAdapter.html) here to avoid having to provide an override for `keyReleased` and `keyTyped`.